### ISSUE TITLE
Issue compiling enclave executables with oelibcxx and IDL files

### ISSUE THREAD

**Issue by pdeligia** [Microsoft/openenclave#176]
It seems that there is an issue when trying to compile an enclave executable (using `add_enclave_executable`) that contains an `.idl` file and trying to link against the enclave C++ runtime (`oelibcxx`).

The issue is that while trying to compile something like the following (using cmake):
```
include(oeidl_file)
include(add_enclave_executable)

oeidl_file(../test.idl enclave gen)

set(SOURCES test.cpp test_t.c)
set(HEADERS test_t.h)

add_enclave_executable(TestEnclave test.conf private.pem
  ${SOURCES} ${gen} ${HEADERS})
target_include_directories(TestEnclave
  PRIVATE ${OPENENCSDK_ROOT}/include ${CMAKE_SOURCE_DIR}/libUtil ${CMAKE_SOURCE_DIR}/libO ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
target_link_libraries(TestEnclave
  SomeCustomCXXLibraryBuiltForEnclave oelibcxx oeidl)
```

It gives the following compile error:
```
In file included from .../openenclave/include/openenclave/libcxx/stdlib.h:88:0,
                 from .../test_t.c:4:
.../openenclave/include/openenclave/libcxx/__config:8:8: error: expected identifier or ‘(’ before string constant
 extern "C" long long strtoll_l(
        ^
.../openenclave/include/openenclave/libcxx/__config:11:8: error: expected identifier or ‘(’ before string constant
 extern "C" unsigned long long int strtoull_l(
        ^
.../openenclave/include/openenclave/libcxx/__config:14:8: error: expected identifier or ‘(’ before string constant
 extern "C" unsigned int arc4random(void);
        ^
```

Our current workaround is to surround the declarations (inside `__config`) with 
```
#ifdef __cplusplus
... // strtoll_l, etc decls
#endif 
```
But this doesn't seem right?

_(And after compiling with our workaround we are getting some issues in the boundaries of calls between Enclave and Host -- seems some arguments might not be initialized properly which causes a segfault -- but we are not sure if it is related to this issue. We are investigating.)_

Interestingly, in the samples, I have seen enclave executables linking against `oelibcxx` (but they do not use idl files) and enclave executables linking against `oelibc` (which use idl files), but there is no sample that does both `oelibcxx` and idl files.

Is the combination of IDL files and oelibcxx not supported yet, or could this just be a bug?

**Comment by CodeMonkeyLeet** [comment]
Resolved by #386, we now use EDL rather than the custom oegen IDL.

**Comment by anakrish** [comment]
Similar issue encountered with EDL as well. The root cause is captured in #127.
Fixed in #985
